### DISCUSSION TITLE
There is a strange space when I use "gb-7714-2015-num" and "font: "Songti SC" with typst 0.11.1

### DISCUSSION THREAD

**Discussion by slgxmh** [typst/typst#4315] [Q&A]
The citation numbers 1, 2, 3 are displayed correctly as [1] [2] [3]. 
<img width="55" alt="image" src="https://github.com/typst/typst/assets/13213238/589c2e1d-5778-4ed6-b47b-474ba97885d5">

However, when the citation number reaches 4, there is an extra space, becoming [4 ].
<img width="79" alt="image" src="https://github.com/typst/typst/assets/13213238/d991966f-aed4-4b4b-ad8d-f3646fade7cb">

Here is my style file:
```typst
#let pageBorder = pattern(
  size: (21cm, 29.7cm),
)[
  #block(inset: 2cm)[
    #rect(
      width: 100%,
      height: 100%,
    )
  ]
]

#let body(body,bib:none)={
  set text(size: 12pt, font: "Songti SC") // here, 
  
  set par(
    justify: true,
    leading: 1.2em,
    first-line-indent: 2em
  )

  set page(
    paper: "a4",
    numbering: "1",
    fill: pageBorder
  )
  
  set heading(numbering: "一、1.1")
  show heading: it=>{
    let levels=counter(heading).get()

    // 一级标题
    if levels.len() == 1 {
      set par(
        first-line-indent: 0em,
      )
      set text(size: 14pt,weight: "bold")
      counter(heading).display(
        numbering("一、",1)
      )
      it.body
      v(0.2em)
      return;
    }

    // 其余标题
    set par(
      first-line-indent: 0em,
    )
    set text(size: 12pt)
    counter(heading).display(
      numbering("1.1.1",1,1,1) 
    )
    it.body
    v(0.5em)
  }

  body

  if bib != none {
    set text(size: 10pt)
    set par(
    justify: true,
    leading: 0.5em,
    first-line-indent: 0em
    )
    set bibliography(style: "gb-7714-2015-numeric", title: none)
    v(1em)
    align(center)[参考文献]
    bibliography(bib)
  }
}
```

And, here is the mian.typ:
```typ
#import "style.typ":*

#show: body.with(bib:"refs.bib")

= 课题来源的项目名称及项目来源
// （属于国家、省、市、自治区、学校、自选中的哪一种？项目编号。）
自选课题

= 立项依据

== 背景和意义

由于人类社会的迅速发展和全球气候环境的不断改变，我国的农业生产面临着严峻的考验。根据联合国的估计，尽管全球人口的增长速度有所减缓，但是在2030年，2050年和2100年，世界人口还将分别达到85亿，97亿和109亿 @desa2019united。在全球所有农产品的需求中，70%的增加依赖于人口的增加，30%依赖于人均收入的增加。因此，到2050年，为了应对不断增加人口和社会需求，世界范围内的农产品产量要增加一倍 @alexandratos2012world。

棉花是世界上最重要的经济作物之一，也是我国最重要的农业种植作物之一。其在国民经济发展中发挥着不可或缺的作用。作为重要的纺织工业原料，国内棉花消费量常年稳定在750万～850万吨，约占全球总量的1/3。近年来我国棉花自给率维持在70%左右。随着国际棉花贸易争端频发，稳定国内生产对于全产业链的健康发展至关重要。相较于传统的三足鼎立格局，当前以新疆棉区为主，黄河流域、长江流域棉区为辅的高度集中化生产布局已经基本成型。统计数据显示，2022年新疆棉花产量539.06万吨，占全国棉花总产量的90.66%。@吴庆君2024中国棉花生产要素替代及区域异质性 我国作为农业大国，要适应国内需求的持续增加，必须要在耕地总量有限的条件下，增加棉花的整体生产。而对作物整个生长周期的信息采集和监控，本质上对商业农场的精细化管理起到指导作业。农作物的检测监控工作，对于田间精细化管理至关重要，可为自动化施药采摘提供有力支持。因此，在棉花成熟阶段的脱叶率和吐絮率检测是精细化田间管理的有效途径。

棉花原产于中南美洲，具有无限生长的习性。与世界其他主要棉花产区想不，新疆地区维度偏高，特别是北疆免去，主要分布在北纬45度附近，无霜期短、秋季降温块是北疆棉区的主要特征。喷施脱叶催熟剂有利于提高采棉机的工作效率，降低棉籽含杂率，是保证棉花顺利采收的重要前提。 @胡莎莎2022脱叶剂对棉花纤维品质 

== 国内外研究现状

随着新疆机采棉的快速发展，棉花在质量上的优势大幅度下降，研究表明，与手采棉相比，机械采收的皮棉降低了2个等级，而且同等级的机采棉质量也低于手摘棉 @王志坚2011新疆机采棉的调研与发展建议 ，严重削弱了新疆在国际市场上原棉质量上的竞争力，对新疆棉花产业持续发展造成一定影响。 @董合忠2014中国棉花种业和原棉品质的国际竞争力分析 大面积种植机采棉后，喷施脱叶催熟剂成为机采棉技术中非常关键的一个环节。如何合理使用脱叶催熟剂

= 课题的研究内容、研究目标、以及拟解决的关键问题。
// （此部分为重点阐述内容）
```

Also, refs.bib:
``` typst
@article{desa2019united,
  title={United Nations Department of economic and social affairs},
  author={Desa, UN},
  journal={Population Division. World Population Prospects},
  year={2019}
}

@article{alexandratos2012world,
  title={World agriculture towards 2030/2050: the 2012 revision},
  author={Alexandratos, Nikos and Bruinsma, Jelle},
  year={2012}
}

@article{吴庆君2024中国棉花生产要素替代及区域异质性,
  title={中国棉花生产要素替代及区域异质性},
  author={吴庆君 and 朱俊峰},
  journal={中国农业大学学报},
  volume={29},
  year={2024}
}

@mastersthesis{胡莎莎2022脱叶剂对棉花纤维品质,
  author = {胡莎莎},
  title = {脱叶剂对棉花纤维品质及种子活力的影响研究},
  school = {石河子大学},
  year = {2022}
}    

@article{王志坚2011新疆机采棉的调研与发展建议,
  title={新疆机采棉的调研与发展建议},
  author={王志坚 and 徐红},
  journal={中国棉花},
  volume={38},
  number={6},
  pages={10--14},
  year={2011}
}

@article{董合忠2014中国棉花种业和原棉品质的国际竞争力分析,
  title={中国棉花种业和原棉品质的国际竞争力分析},
  author={董合忠},
  journal={中国棉麻流通经济},
  number={4},
  pages={31--37},
  year={2014}
}
```

**Comment by istudyatuni** [comment]
Please use English

**Reply by slgxmh** [reply]
I have translated it to English and thank to you remind.

**Comment by csimide** [comment]
I was unable to reproduce this situation. Could you please provide the code?


---

translation for who might concern:

The citation numbers 1, 2, 3 are displayed correctly as `[1]` `[2]` `[3]`. However, when the citation number reaches 4, there is an extra space, becoming `[4 ]`.

**Reply by slgxmh** [reply]
I have translated it to English and pasted the code, thank to your concern.

**Reply by Coekjan** [reply]
Could you paste your `refs.bib` file?

**Reply by slgxmh** [reply]
Thanks again for the reminder, I've pasted refs.bib.

**Comment by Coekjan** [comment]
I tested the code with released prebuilt binaries (v0.11.0 and v0.11.1) and online app, and I could not reproduce the bug.

**Reply by slgxmh** [reply]
Thanks for your test. I tested the code with online app too, I have found that the key to issue is the font.

The space disappeared when I changed font to "Source Han Serif".

**Reply by Coekjan** [reply]
Maybe it relate with https://github.com/typst/typst/issues/4006 . You could try the workaround: https://github.com/typst/typst/issues/4006#issuecomment-2076671145

**Comment by csimide** [comment]
**TL; DR:** Adding the following code may solve the problem:
```typst
#show ref.where(element: none): set text(font: ("STSong SC", "STSong"))
// I noticed you're using Songti SC, which suggests you're on macOS, so I'm using STSong SC here
```

Alternatively, you can change the text font from Songti SC to STSong SC.

---

I suspect the issue arises because the reference numbers use superscript characters `¹` (U+00B9), `²` (U+00B2), and `³` (U+00B3) for 1, 2, and 3, respectively, while numbers 4 and above use regular digits with superscript styling. However, this can lead to inconsistent rendering across different fonts. In SimSun, this causes extra spaces after 1, 2, and 3, while numbers 4 and above appear correctly. The following code can be used for testing:

```typst
1-3:
#super[\[]¹#super[\]]
#super[\[]²#super[\]]
#super[\[]³#super[\]]

>=4:
#super[\[4\]]
#super[\[5\]]
#super[\[6\]]
```

![image](https://github.com/typst/typst/assets/117500413/e93ee020-ebdb-4aa9-a535-c3c298d295af)

**Comment by Enivex** [comment]
If the issue is what @csimide describes, then disabling the `typographic` option on text should be the simplest solution 

Relevant: https://github.com/typst/typst/issues/4006
